src/osc.js

```diff
@@ -208,20 +208,20 @@ var osc = osc || {};
      */
     osc.writeString = function (str) {
         var terminated = str + "\u0000",
-            len = terminated.length,
-            paddedLen = (len + 3) & ~0x03,
-            arr = new Uint8Array(paddedLen);
-
         var encoder = osc.isBufferEnv ? osc.writeString.withBuffer :
             osc.TextEncoder ? osc.writeString.withTextEncoder : null,
             encodedStr;
 
         if (encoder) {
-            encodedStr = encoder(terminated);
+            terminated = encoder(terminated);
         }
 
+        len = terminated.length,
```

Comment:
In JavaScript, assigment to variables that haven't been defined with ```var``` (or the preferable ```let``` statement, but osc.js' largely predates it) are implicitly defined as global variables. We should ensure they're properly defined here.